File: SpringBackend/src/main/java/com/HomeHive/service/ComplaintService.java
package com.HomeHive.service;

import java.util.List;

import com.HomeHive.entities.Complaint;
import com.HomeHive.enums.ComplaintStatus;

public interface ComplaintService {
	Complaint raiseComplaint(Complaint complaint);
	
	List<Complaint> getMyComplaints();
	
	List<Complaint> getAllComplaints();
	
	Complaint updateComplaintStatus(Long complaintId, ComplaintStatus status, String adminResponse);
}

File: ReactFrontend/src/Components/Services/ComplaintService.js
import axios from 'axios';
import { getToken } from './UserService';

const BASE_URL = 'http://localhost:8080/complaints';


export function raiseComplaint(complaintData) {
  return axios.post(`${BASE_URL}/raise-complaint`, complaintData, {
    headers: {
      Authorization: `Bearer ${getToken()}`,
    },
  });
}


export function getMyComplaints() {
  return axios.get(`${BASE_URL}/my-complaints`, {
    headers: {
      Authorization: `Bearer ${getToken()}`,
    },
  });
}


export function getAllComplaints() {
  return axios.get(`${BASE_URL}/all-complaints`, {
    headers: {
      Authorization: `Bearer ${getToken()}`,
    },
  });
}


export function updateComplaintStatus(complaintId, statusData) {
  return axios.put(`${BASE_URL}/${complaintId}/status`, statusData, {
    headers: {
      Authorization: `Bearer ${getToken()}`,
    },
  });
}
